tests/test_magnetic_fields.py

```diff
@@ -740,3 +740,22 @@ def test_Bnormal_save_and_load_HELIOTRON(self, tmpdir_factory):
         asym_surf = FourierRZToroidalSurface(sym=False)
         with pytest.raises(AssertionError, match="sym"):
             Bnorm_from_file = read_BNORM_file(path, asym_surf, grid)
+
+    @pytest.mark.unit
+    def test_spline_field_jit(self):
+        """Test that the spline field can be passed to a jitted function."""
+        field = SplineMagneticField.from_mgrid
+
+        extcur = [4700.0, 1000.0]
+        mgrid = "tests/inputs/mgrid_test.nc"
+        field = SplineMagneticField.from_mgrid(mgrid, extcur)
+
+        x = jnp.array([0.70, 0, 0])
+
+        @jit
+        def foo(field, x):
+            return field.compute_magnetic_field(x)
+
+        np.testing.assert_allclose(
+            foo(field, x), np.array([[0, -0.671, 0.0858]]), rtol=1e-3, atol=1e-8
+        )
```

Comment:
those values are the "correct" values for this field at that point, they're also used in another test above. The real thing being tested here is that the field object itself can be passed to a jitted function without throwing an error.

Reply:
I guess I'm asking how we know those are the "correct" values. Do we know it from theory? If not I don't think they should be hard-coded. This is more a complaint about the other test you copied them from, but would be good to fix it now.

Reply:
The x location is a node in the mgrid file, so we know the field "exactly" (or as exactly as mgrid made it). The point of hardcoding that as opposed to doing some indexing etc is that this was to test if I did the indexing/reshaping etc correctly in `SplineMagneticField`, otherwise we'd just be testing something like `assert B[idx] == B[idx]`

Reply:
Ah ok that makes sense. Thanks